implementations/rust/ockam/ockam_api/src/nodes/service/secure_channel.rs

```diff
@@ -488,20 +487,19 @@ impl NodeManagerWorker {
         let addr = Address::from(body.addr.as_ref());
         info!(%addr, "Handling request to delete secure channel listener");
         let mut node_manager = self.node_manager.write().await;
-        let res = match node_manager
-            .delete_secure_channel_listener_impl(&addr)
-            .await
-        {
-            Ok(()) => {
+        let id = req.id();
+        let mut payload: Option<Address> = None;
+        Ok(match node_manager.delete_secure_channel_listener_impl(&addr).await {
+            Some(_) => {
                 trace!(%addr, "Removed secure channel listener");
-                Some(addr)
+                payload = Some(addr);
+                Response::ok(id)
             }
-            Err(err) => {
-                trace!(%addr, %err, "Error removing secure channel listener");
-                None
+            None => {
+                trace!(%addr, "Error removing secure channel listener");
+                Response::not_found(id)
```

Comment:
I think this change here is all you need to check the `NotFound` error on the command side. 

You don't even need to put the payload behind an `Option` because you can check the response code from the header directly. This is done automatically when you call the ` rpc.is_ok()?;` function. If that call doesn't return an error, then you are good to parse the response body.

Reply:
Yes, that approach is nicer. I consistently went the down route taken for the "show" subcomand; 0214630